test/parallel/test-http-parser-finish-error.js

```diff
@@ -0,0 +1,28 @@
+'use strict';
+
+const common = require('../common');
+const net = require('net');
+const http = require('http');
+const assert = require('assert');
+
+const str = 'GET / HTTP/1.1\r\n' +
+            'Content-Length:';
+
+
+const server = http.createServer(common.mustNotCall());
+server.on('clientError', common.mustCall((err, socket) => {
+  assert(/^Parse Error/.test(err.message));
+  assert.strictEqual(err.code, 'HPE_INVALID_EOF_STATE');
+  server.close();
+  socket.destroy();
+}, 1));
+server.listen(0, () => {
+  const client = net.connect({ port: server.address().port }, () => {
+    client.on('data', common.mustNotCall());
+    client.on('end', common.mustCall(() => {
+      server.close();
```

Comment:
Just a nit that ca be ignored: we are calling this twice, the one in the `'clientError'` listener is sufficient.

Reply:
Ack